megatron/tokenizer/tokenizer.py

```diff
@@ -282,9 +297,10 @@ def __init__(self, vocab_file=None, fast=True):
         else:
             self.tokenizer = GPT2Tokenizer.from_pretrained(vocab_name)
 
-        self.tokenizer.add_special_tokens({"pad_token": "<|padding|>"})
+        self.tokenizer.add_special_tokens({"pad_token": "<|padding|>","eos_token":"<|endoftext|>"})
```

Comment:
What is the vocab already contains eod?

Reply:
gpt2tokenzer indead has eos_token and that exactly is "<|endoftext|>". 
We can remove add "eos_token" part but I do this explictly to show what eos_token we are using.

Reply:
I see, I would say we should remove it, because someone might use a different vocab where the eos token is different?